recipes-bsp/u-boot/files/0010-board-siemens-iot2050-get-and-set-m.2-releated-infor.patch

```diff
@@ -0,0 +1,223 @@
++		pr_err("IOT2050:Can not set the pin value\n");
++		return ;
++	}
++	dm_gpio_set_value(&gpio,value);
++}
++
++static m2_bkey_config_t get_m2_config(void)
++{
++	m2_bkey_config_t result;
++
++	result.config_0 = get_pinvalue(m2_key_config[0].gpio_name, m2_key_config[0].label);
++	result.config_1 = get_pinvalue(m2_key_config[1].gpio_name, m2_key_config[1].label);
++	result.config_2 = get_pinvalue(m2_key_config[2].gpio_name, m2_key_config[2].label);
++	result.config_3 = get_pinvalue(m2_key_config[3].gpio_name, m2_key_config[3].label);
++
++	return result;
++}
++
++static serdes_select_t get_key_config_set_m2_pinmux(void)
++{
++	int i;
++	serdes_select_t serdes_select = {USB30, PCIE1_LANE0}; /* default setting */
++	int config_table_size= sizeof(config_table)/sizeof(config_table[0]);
++	m2_bkey_config_t config = get_m2_config();
++
++	printf("B-KEY:");
++	printf("%d",config.config_0);
++	printf("%d",config.config_1);
++	printf("%d",config.config_2);
++	printf("%d",config.config_3);
++	printf("\n");
++
++	for(i = 0; i < config_table_size; i++) {
++		if(!memcmp(&config, &config_table[i].m2_bkey_config, sizeof(m2_bkey_config_t))) {
++			/* set pinmux */
++			set_pinvalue(m2_pinmux_config[0].gpio_name, m2_pinmux_config[0].label, config_table[i].m2_pinmux_select.sel_usb30_pcie0_lane0);
++			set_pinvalue(m2_pinmux_config[1].gpio_name, m2_pinmux_config[1].label, config_table[i].m2_pinmux_select.sel_pcie1_pcie0);
++			set_pinvalue(m2_pinmux_config[2].gpio_name, m2_pinmux_config[2].label, config_table[i].m2_pinmux_select.sel_usb30_pcie0_lane1);
++			/* get serdes config info */
++			serdes_select =  config_table[i].serdes_select;
++			break;
++		}
++	}
++
++	return serdes_select;
++}
++
++/*
++ * M.2 Board is a Variant Baesed On PG2 board
++ */
++static bool board_is_m2(void)
++{
++	struct iot2050_info *info = IOT2050_INFO_DATA;
++
++	return info->magic == IOT2050_INFO_MAGIC &&
++		strstr((char *)info->name, "-M2") != NULL;
++}
++
++static void configure_serdes_by_overlays(void)
++{
++	struct iot2050_info *info = IOT2050_INFO_DATA;
++	serdes_select_t serdes_sel;
++
++	/* enable m.2 connector power */
++	set_pinvalue(m2_enable_power.gpio_name, m2_enable_power.label, 1);
++	udelay(4 * 100);
++
++	if(env_get("m2_manual_config") && env_get("m2_overlays")) {
++		printf("Maunal Select M.2 Configuration\n");
++	}
++	else
++	{
++		serdes_sel = get_key_config_set_m2_pinmux();
++		if (serdes_sel.lane0_sel == PCIE0_LANE0 && serdes_sel.lane1_sel == PCIE0_LANE1)
++			env_set("m2_overlays", "ti/k3-am654-iot2050-bkey-pciex2.dtbo");
++		else if (serdes_sel.lane0_sel == USB30 && serdes_sel.lane1_sel == PCIE0_LANE1)
++			env_set("m2_overlays", "ti/k3-am654-iot2050-bkey-usb3-ekey-pcie.dtbo");
++		else if (serdes_sel.lane0_sel == PCIE0_LANE0 && serdes_sel.lane1_sel == PCIE1_LANE0)
++			env_set("m2_overlays", "ti/k3-am654-iot2050-bkey-pcie-ekey-pcie.dtbo");
++		else
++			env_set("m2_overlays", "ti/k3-am654-iot2050-bkey-usb3-ekey-pcie.dtbo");
```

Comment:
`m2_overlays` - or just `m2_overlay`? Means, will we eventually have to provide multiple overlays via this variable? Just for consistency.